enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -966,25 +1024,58 @@ def get_group_member_data_with_aggregates(self, request, uuid):
+            # ie set `traverse_pagination` to True
+            sort_by = None
+            request_traverse_pagination = traverse_pagination
+            traverse_pagination = True
+            request_page = page
+            page = None
+
         # Request the group member data from platform
         member_response = LmsApiClient().fetch_group_members(
             group_uuid=group_uuid,
-            sort_by=request_serializer.validated_data.get('sort_by'),
+            sort_by=sort_by,
             user_query=request_serializer.validated_data.get('user_query'),
             show_removed=request_serializer.validated_data.get('show_removed'),
-            is_reversed=request_serializer.validated_data.get('is_reversed'),
+            is_reversed=is_reversed,
             traverse_pagination=traverse_pagination,
             page=page,
         )
         member_results = member_response.get('results')
+
         # Sift through the group members data, zipping the aggregate data from the subsidy service into
         # each member record, assume enrollment count is 0 if subsidy enrollment aggregate data does not exist.
-        for key, result in enumerate(member_results):
-            enrollment_count = 0
-            if lms_user_id := result.get('lms_user_id'):
-                enrollment_count = subsidy_learner_aggregate_dict.get(lms_user_id, 0)
-            result['enrollment_count'] = enrollment_count
-            member_results[key] = result
+        member_results = zip_group_members_data_with_enrollment_count(member_results, subsidy_learner_aggregate_dict)
+
+        # If the request sorts by enrollment count, sort member data and restore the request variables
+        # so we can properly construct the `next` and `previous` pages
+        page_index_start = 0
+        num_member_results = 0
+        if sort_by_enrollment_count:
+            member_results.sort(key=lambda result: result.get('enrollment_count'), reverse=(not is_reversed))
+            # Only if the request asked for a specific page of data do we need to restore page and traverse_pagination
+            if request_page:
+                # Needed to construct `next` and `previous` values for the response
+                num_member_results = len(member_results)
+
+                # Cut down the returned "all data" to the page and size requested
+                page_index_start = (request_page - 1) * GROUP_MEMBERS_WITH_AGGREGATES_DEFAULT_PAGE_SIZE
+                member_results = member_results[
+                    page_index_start: page_index_start + GROUP_MEMBERS_WITH_AGGREGATES_DEFAULT_PAGE_SIZE
+                ]
+
+                # Restore the values
+                page = request_page
+                # No need to restore `traverse_pagination` if request_page is None because traverse must be true
+                # if no page is supplied
+                traverse_pagination = request_traverse_pagination
```

Comment:
This is a code smell pointing at the fact that you're mutating query param values on the way in for the sake of making requests to the `fetch_group_members` endpoint. Instead of doing that, why don't you compute a different set of variables used only for that call? And even better would be to extract a helper method to both compute those variables and make the call?